packages/grid/_modules_/grid/components/GridRowCount.tsx

```diff
@@ -4,22 +4,28 @@ import { GridApiContext } from './GridApiContext';
 
 interface RowCountProps {
   rowCount: number;
+  visibleRowCount: number;
 }
 
 type GridRowCountProps = React.HTMLAttributes<HTMLDivElement> & RowCountProps;
 
 export const GridRowCount = React.forwardRef<HTMLDivElement, GridRowCountProps>(
   function GridRowCount(props, ref) {
-    const { className, rowCount, ...other } = props;
+    const { className, rowCount, visibleRowCount, ...other } = props;
     const apiRef = React.useContext(GridApiContext);
 
     if (rowCount === 0) {
       return null;
     }
 
+    const text =
+      visibleRowCount < rowCount
+        ? apiRef!.current.getLocaleText('footerTotalVisibleRows')(visibleRowCount, rowCount)
+        : rowCount.toLocaleString();
+
     return (
       <div ref={ref} className={clsx('MuiDataGrid-rowCount', className)} {...other}>
-        {`${apiRef!.current.getLocaleText('footerTotalRows')} ${rowCount.toLocaleString()}`}
+        {`${apiRef!.current.getLocaleText('footerTotalRows')} ${text}`}
```

Comment:
How about we do:
```suggestion
        {apiRef!.current.getLocaleText('footerTotalRows')}{' '}{text}
```

to avoid the string casting and so we can stay true to the type of the key. For instance: https://github.com/mui-org/material-ui-x/blob/4a57468f36acb3be4910693df733c0c1f5a20f85/packages/grid/_modules_/grid/models/api/gridLocaleTextApi.ts#L85

cc @DanailH who wrote this line initially.